src/include/units/quantity.h

```diff
@@ -48,6 +48,158 @@ concept safe_divisible = // exposition only
     treat_as_floating_point<Rep> ||
     ratio_divide<typename UnitFrom::ratio, typename UnitTo::ratio>::is_integral();
 
+//  constrain operators to mpusz::units::quantity only
+template<typename T>
+concept ll_quantity_concept = detail::is_quantity<T>;
+
+// base class for quantity
+// no increase in size of derived object , see https://en.cppreference.com/w/cpp/language/ebo
+struct quantity_friend_operations{
```

Comment:
This actually makes the interface way more complicated and I do not know any other class int the Standard Library (or others) that follow this idiom. I do not want to introduce anything strange to this class template because every such change will lower the probability of acceptance by the ISO C++ Committee.